Assignment/Program.c

```diff
@@ -9,33 +9,113 @@
+            fwprintf (output, L"%lc", blackPawn);
+         }
+         else if (i == 6) {
+            wprintf (L"%lc ", whitePawn);
+            fwprintf (output, L"%lc ", whitePawn);
+         }
+         else if (i == 7) {
+            wprintf (L"%lc ", whitePieces[j]);
+            fwprintf (output, L"%lc ", whitePieces[j]);
+         }
+         else {
+            wprintf (L"  ");
+            fwprintf (output, L"  ");
+         }
       }
       wprintf (L"%lc\n", 0x2503);
+      fwprintf (output, L"%lc\n", 0x2503);
       if (i < 7) {
          wprintf (L"%lc", 0x2523);
-         for (int k = 0; k < 7; k++) wprintf (L"%lc%lc%lc%lc", 0x2501, 0x2501, 0x2501, 0x254B);
+         fwprintf (output, L"%lc", 0x2523);
+         for (int k = 0; k < 7; k++) {
+            wprintf (L"%lc%lc%lc%lc", 0x2501, 0x2501, 0x2501, 0x254B);
+            fwprintf (output, L"%lc%lc%lc%lc", 0x2501, 0x2501, 0x2501, 0x254B);
+         }
          wprintf (L"%lc%lc%lc%lc\n", 0x2501, 0x2501, 0x2501, 0x252B);
+         fwprintf (output, L"%lc%lc%lc%lc\n", 0x2501, 0x2501, 0x2501, 0x252B);
       }
    }
-   wprintf (L"%lc", 0x2517); //bottom
-   for (int i = 0; i < 7; i++) wprintf (L"%lc%lc%lc%lc", 0x2501, 0x2501, 0x2501, 0x253B);
+   // bottom border
+   wprintf (L"%lc", 0x2517);
+   fwprintf (output, L"%lc", 0x2517);
+   for (int i = 0; i < 7; i++) {
+      wprintf (L"%lc%lc%lc%lc", 0x2501, 0x2501, 0x2501, 0x253B);
+      fwprintf (output, L"%lc%lc%lc%lc", 0x2501, 0x2501, 0x2501, 0x253B);
+   }
    wprintf (L"%lc%lc%lc%lc", 0x2501, 0x2501, 0x2501, 0x251B);
-   return 0;
+   fwprintf (output, L"%lc%lc%lc%lc", 0x2501, 0x2501, 0x2501, 0x251B);
+}
+
+int CompareFiles (const char* outputFile, const char* referenceFile) {
+   FILE* output = fopen (outputFile, "r, ccs=UTF-8"),
+      * reference = fopen (referenceFile, "r, ccs=UTF-8");
+   if (output == NULL || reference == NULL) {
+      wprintf (L"Error opening files for comparison.\n");
+      return -1;
+   }
+   int row = 1, col = 1;
+   wchar_t outputChar, referenceChar;
+   while (1) {
+      outputChar = fgetwc (output);
+      referenceChar = fgetwc (reference);
+      if (outputChar == WEOF && referenceChar == WEOF) {
+         wprintf (L"\n\nTest Passed\n");
+         fclose (output);
+         fclose (reference);
+         return 0;
+      }
+      if (outputChar != referenceChar) {
+         wprintf (L"\nMismatch at row %d, column %d\n", row, col);
+         fclose (output);
+         fclose (reference);
+         return -2;
+      }
+      if (outputChar == L'\n' || referenceChar == L'\n') {
+         row++;
+         col = 1;
+      }
+      else col++;
+   }
+}
+
+int main () {
+   int a = _setmode (_fileno (stdout), _O_U8TEXT);
+   FILE* output = fopen ("output.txt", "w, ccs=UTF-8");
+   if (output == NULL) {
+      wprintf (L"Failed to open output file.\n");
+      return -1;
+   }
+   PrintChessBoard (output);
+   fclose (output);
+   int result = CompareFiles ("output.txt", "ref.txt");

```

Comment:
since result variable is used once, no need to have it directly return the output of Compare files method